### PR TITLE
Fix content extraction

### PR DESCRIPTION
**Pull request by abhishekvm**
----
https://www.nytimes.com/2017/05/17/us/politics/robert-mueller-special-counsel-russia-investigation.html
- Extra text was getting fetched in article body
- Added rules in setToRemove to exclude unwanted elements

----
http://nymag.com/daily/intelligencer/2017/04/march-jobs-report-falls-short-of-expectations.html
- Quoted text was not extracted
- Added rule in nodesToKeepCssSelector

----
http://blog.dogsbite.org/2014/08/014-dog-bite-fatality-toddler-dies-family-pit-bull-attack-under-grandmothers-care.html
- Ignored wrong TestCase

----
http://www.delish.com/cooking/g2668/spring-asparagus-dishes/
http://www.delish.com/cooking/g2669/breakfast-casserole-recipes/
- Assign the description as a content if article doen't have content at all

### CODE REVIEW

File: src/main/java/de/jetwick/snacktory/ArticleTextExtractor.java [hunk 1]
```diff
@@ -926,6 +932,40 @@ protected Date extractDate(Document doc) {
...
+                        return parsedDate;
+                    }
+                }
+            }
+        }
+
+        // http://www.business-money.com/announcements/catalyst-achieves-300m-funding-milestone
+        // There is no metadata at all to identify the date other than complete css selector
+        // This might be really bad hack but we don't have other option
+        elems = doc.select("body > div:nth-child(7) > div > div.col-sm-9 > div > h1 > span > p > span");
```
Comment (andresp99999) [inline src/main/java/de/jetwick/snacktory/ArticleTextExtractor.java]: I wonder if we can do as a last resort some sort of search for common dates format in the article body. Ideally we don't to get this specific when defining rules (although in some cases we may have not option).

Comment (skyshard) [reply src/main/java/de/jetwick/snacktory/ArticleTextExtractor.java]: yes, this is way too specific/brittle of a selector, we might not want to merge this in favor of that date checking rule instead

Comment (skyshard) [reply src/main/java/de/jetwick/snacktory/ArticleTextExtractor.java]: let's move this specific case out, the rest can be merged

**Review by skyshard — CHANGES_REQUESTED** [review]

**Comment by abhishekvm** [comment]
@skyshard Addressed review comments. Consider merging this if no more action item.

**Pull request merged**